app/controllers/service_provider_controller.rb

```diff
@@ -2,11 +2,27 @@ class ServiceProviderController < ApplicationController
   protect_from_forgery with: :null_session
 
   def update
-    if FeatureManagement.use_dashboard_service_providers?
-      ServiceProviderUpdater.new.run
-      SecureHeadersWhitelister.new.run
+    authorize do
+      if FeatureManagement.use_dashboard_service_providers?
+        ServiceProviderUpdater.new.run
+        SecureHeadersWhitelister.new.run
+      end
+
+      render json: { status: 'If the feature is enabled, service providers have been updated.' }
+    end
+  end
+
+  private
+
+  def authorize
+    if authorization_token == Figaro.env.dashboard_api_token
+      yield
+    else
+      render nothing: true, status: 401
     end
+  end
 
-    render json: { status: 'If the feature is enabled, service providers have been updated.' }
+  def authorization_token
+    request.headers['X-LOGIN-DASHBOARD-TOKEN']
```

Comment:
https://github.com/rack/rack/blob/0d74df67db33cc2a9858ea1568bf47ad8583d84c/SPEC#L54-L65

which links to the CGI spec: https://tools.ietf.org/html/rfc3875#section-4.1.18

which has this blurb: 

> The HTTP header field name is converted to upper case, has all
>   occurrences of "-" replaced with "\_" and has "HTTP\_" prepended to
>   give the meta-variable name.